clvm-traits/src/error.rs

```diff
@@ -1,34 +1,32 @@
-use clvmr::{allocator::NodePtr, reduction::EvalErr};
-use thiserror::Error;
+use std::string::FromUtf8Error;
 
-#[derive(Debug, Clone, PartialEq, Eq, Error)]
-pub enum Error {
-    #[error("allocator error {0:?}")]
-    Allocator(EvalErr),
+use thiserror::Error;
 
-    #[error("expected atom")]
-    ExpectedAtom(NodePtr),
+/// Any errors that may occur while converting a Rust value to a CLVM value.
+#[derive(Error, Debug, Clone, PartialEq, Eq)]
+pub enum ToClvmError {
+    #[error("limit reached")]
+    LimitReached,
```

Comment:
it would be less generic and more concrete (I think) to call this something like out of memory. As far as I can tell, it's only used when the Allocator hits its memory limit